src/Components/hooks/useMalwareSearchParams.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+const useMalwareSearchParams = (malwareSearchParams, stateSet) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const handleSetSearchParams = (newParams) => {
+    let newSearchParams = {};
+    newParams.forEach((param) => {
+      param.value?.length
+        ? (newSearchParams[param.name] = param.value)
+        : undefined;
+    });
+
+    setSearchParams(newSearchParams);
+  };
+
+  useEffect(() => {
+    let payload = {};
+
+    malwareSearchParams.forEach((param) => {
+      if (param.type === 'string') {
+        payload[param.name] = searchParams.get(param.name) || '';
+      } else if (param.type === 'array') {
+        searchParams.getAll(param.name).length
+          ? (payload[param.name] = searchParams.getAll(param.name))
+          : (payload[param.name] = undefined);
+      }
+    });
+
+    stateSet({
+      type: 'setTableVars',
+      payload: { ...payload, offset: 0 },
+    });
+  }, [searchParams]);
```

Comment:
Since this `useEffect` accesses `malwareSearchParams`, it should also be a dependency here. The same for `stateSet`. However, to be able to properly make this work, we might be better off to extract the logic into a memoised function and make that a dependency instead. That would look something like this:

```js
  const setSearchParamState = useCallback((searchParams) => {
    let payload = {};

     malwareSearchParams.forEach((param) => {
       if (param.type === 'string') {
         payload[param.name] = searchParams.get(param.name) || '';
       } else if (param.type === 'array') {
         searchParams.getAll(param.name).length
           ? (payload[param.name] = searchParams.getAll(param.name))
           : (payload[param.name] = undefined);
       }
     });

     stateSet({
       type: 'setTableVars',
       payload: { ...payload, offset: 0 },
     });
  }, [JSON.stringify(malwareSearchParams), stateSet])

   useEffect(() => {
    setSearchParamState(searchParams)
   }, [searchParams, setSearchParamState]);
```